packages/bruno-app/src/components/RequestPane/Auth/OAuth1/index.js

```diff
@@ -0,0 +1,131 @@
+import React, { forwardRef, useCallback, useRef } from 'react';
+import get from 'lodash/get';
+import { useTheme } from 'providers/Theme';
+import { useDispatch } from 'react-redux';
+import SingleLineEditor from 'components/SingleLineEditor';
+import { updateAuth } from 'providers/ReduxStore/slices/collections';
+import { sendRequest, saveRequest } from 'providers/ReduxStore/slices/collections/actions';
+import StyledWrapper from './StyledWrapper';
+import { inputsConfig } from './inputsConfig';
+import { useTranslation } from 'react-i18next';
+import Dropdown from 'components/Dropdown';
+import { IconCaretDown } from '@tabler/icons';
+import FilePickerEditor from 'components/FilePickerEditor';
+import path from 'path';
+import { isWindowsOS } from 'utils/common/platform';
+import slash from 'utils/common/slash';
+
+const OAuth1 = ({ item, collection }) => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { storedTheme } = useTheme();
+
+  const oAuth1 = item.draft ? get(item, 'draft.request.auth.oauth1', {}) : get(item, 'request.auth.oauth1', {});
+
+  const handleRun = () => dispatch(sendRequest(item, collection.uid));
+  const handleSave = () => dispatch(saveRequest(item.uid, collection.uid));
+
+  const refs = useRef(new Map());
+  const setRef = useCallback((ref, key) => {
+    if (ref) {
+      refs.current.set(key, ref);
+    } else {
+      refs.current.delete(key);
+    }
+  }, []);
+
+  const hideDropdown = (key) => {
+    const dropdown = refs.current.get(key);
+    if (dropdown?.hide) {
+      dropdown.hide();
+    }
+  };
+
+  const handleChange = (key, val) => {
+    console.log(key, val);
+    dispatch(
+      updateAuth({
+        mode: 'oauth1',
+        collectionUid: collection.uid,
+        itemUid: item.uid,
+        content: {
+          ...oAuth1, [key]: val
+        }
+      })
+    );
+  };
+
+  const relativeFile = (file) => {
+    if (file) {
+      if (isWindowsOS()) {
+        return slash((path.win32.relative(collection.pathname, file)));
+      } else {
+        return path.posix.relative(collection.pathname, file);
+      }
+    } else {
+      return '';
+    }
+  };
```

Comment:
This may be outdated method of getting relative path to a local file with recent changes to file picker, not sure.